File: 739. Daily Temperatures/daily_temperatures.rs
impl Solution {
    
    // O(n) time,
    // O(n) space,
    // Approach: mono stack, 
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let temp_len = temperatures.len();
        let mut answer = vec![0;temp_len];
        let mut stack = Vec::new();
        
        for i in 0..temp_len {   
            while stack.len() > 0 && &temperatures[i] > &temperatures[stack[stack.len()-1]] {
                let popped = stack.pop().unwrap();
                answer[popped] = (i-popped) as i32;
            }
            stack.push(i);
        }
        
        return answer;
    }
}

File: 739. Daily Temperatures/DailyTemperatures.go
// O(n) time,
// O(n) space,
// Approach: mono stack, 
func dailyTemperatures(temperatures []int) []int {
    var n int = len(temperatures);
    answer:= make([]int, n);
    var stack []int;
    
    for i:= 0; i < n; i++ {
        for len(stack) > 0 && temperatures[i] > temperatures[stack[len(stack)-1]] {
            popped:= stack[len(stack)-1];
            stack = stack[:len(stack)-1];
            answer[popped] = i-popped;
        }
        stack = append(stack, i);
    }
    
    return answer;
}